code/modules/mob/living/silicon/damage_procs.dm

```diff
@@ -49,3 +49,16 @@
 		return ..()
 
 	return FALSE
+
+/mob/living/silicon/adjustFireLoss(amount, updating_health = TRUE, forced = FALSE) //immune to oxygen damage
```

Comment:
```suggestion
/mob/living/silicon/ai/adjustFireLoss(amount, updating_health = TRUE, forced = FALSE) //immune to oxygen damage
```
Would this not eliminate the need for a type check?

Reply:
true